packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/ClientEventService.java

```diff
@@ -63,28 +62,48 @@ public static class UserClientEvent<T> implements Serializable {
 		private ClientEvent<T> event;
 	}
 
-	@PostConstruct
-	void init() {
-		allUsersQueue = new Queue(CLIENT_ALL_USERS_EVENT_QUEUE, config.getDurableQueues(), false, false);
-		rabbitAdmin.declareQueue(allUsersQueue);
-
-		userQueue = new Queue(CLIENT_USER_EVENT_QUEUE, config.getDurableQueues(), false, false);
-		rabbitAdmin.declareQueue(userQueue);
-	}
-
 	/**
 	 * Connects a user to the SSE service
 	 *
 	 * @param user the user to connect
 	 * @return the emitter to send messages to the user
 	 */
 	public SseEmitter connect(final User user) {
-		final SseEmitter emitter = new SseEmitter();
-		if (!userIdToEmitter.containsKey(user.getId())) {
-			userIdToEmitter.put(user.getId(), new ArrayList<>());
+		synchronized (userIdToEmitters) {
+			final List<SseEmitter> emitters = userIdToEmitters.getOrDefault(user.getId(), new ArrayList<>());
+			final SseEmitter emitter = new SseEmitter();
+			emitter.onError((e) -> {
+				emitter.complete();
+				synchronized (userIdToEmitters) {
+					emitters.remove(emitter);
+					if (emitters.isEmpty()) {
+						userIdToEmitters.remove(user.getId());
+					}
+				}
+			});
+			emitters.add(emitter);
+			userIdToEmitters.put(user.getId(), emitters);
```

Comment:
We should probably be using a more thread safe data model instead of the ArrayList internally, perhaps the ConcurrentLinkedQueue<> would be a smarter choice within the ConcurrentHashMap<>.

Reply:
fixed